modules/mod_facl.c

```diff
@@ -1124,39 +1143,59 @@ static int facl_fsio_access(pr_fs_t *fs, const char *path, int mode,
 
       if (sys_access(fs, path, mode, uid, gid, suppl_gids) < 0) {
         xerrno = errno;
+
         pr_trace_msg(trace_channel, 6, "normal access check for '%s' "
           "failed: %s", path, strerror(xerrno));
         errno = xerrno;
         return -1;
       }   
 
       return 0;
-
-    } else {
-      return -1;
     }
+
+    errno = xerrno;
+    return -1;
+  }
+# endif
+
+  if (tmp_pool == NULL) {
+    tmp_pool = make_sub_pool(fs->fs_pool);
+    pr_pool_tag(tmp_pool, "mod_facl access(2) pool");
   }
+
+  res = check_facl(tmp_pool, path, mode, acls, nents, &st, uid, gid,
+    suppl_gids);
+  xerrno = errno;
+
+# if defined(HAVE_BSD_POSIX_ACL) || \
+     defined(HAVE_LINUX_POSIX_ACL) || \
+     defined(HAVE_MACOSX_POSIX_ACL)
+  acl_free(acls);
```

Comment:
And this call to `acl_free(3)` is the other main source of the memory leak here.